libV2/vault.js

```diff
@@ -83,14 +84,29 @@ class Vault {
                         reject(err);
                         return;
                     }
-                    resolve(res);
+                    if (!res.message || !res.message.body) {
+                        reject(errors.InternalError);
```

Comment:
should this return or an else for resolve? I don't think promises allow resolve or reject more than once though.

Reply:
yep missing a return